arch/arm64/core/mmu.c

```diff
@@ -629,7 +630,17 @@ static int __add_map(struct arm_mmu_ptables *ptables, const char *name,
 	__ASSERT(((virt | phys | size) & (CONFIG_MMU_PAGE_SIZE - 1)) == 0,
 		 "address/size are not page aligned\n");
 	desc |= phys;
-	return set_mapping(ptables, virt, size, desc, may_overwrite);
+	ret = set_mapping(ptables, virt, size, desc, may_overwrite);
+#ifdef CONFIG_DT_HAS_XEN_XEN_ENABLED
+	if (!ret) {
+		unsigned int mem_type = MT_TYPE(attrs);
+
+		if ((mem_type == MT_NORMAL) || (mem_type == MT_NORMAL_NC)) {
```

Comment:
`NC` stands non-cacheable, right? Should it be invalidated? I am more interested in `WB` mappings